### DISCUSSION TITLE
Return type for `get_child_by_id` and `get_widget_by_id`

### DISCUSSION THREAD

**Discussion by aaronst** [Textualize/textual#1237] [Q&A] [answered]
Loving the new `get_child_by_id` and `get_widget_by_id` methods! But I've noticed that Pylance gets real unhappy with the return type of `Widget`. For instance, if I get a `Static` and then call `update`. Is there any `typing` magic we can add to make this better, or is this something we'll just need to ignore?

**Comment by willmcgugan** [comment, marked as answer]
We should do something similar to query, where you pass on the type and it validates it at runtime and statically. 

In the meantime try an `assert isinstance(widget, MyWidget)` to quell PyLance.

**Reply by aaronst** [reply]
I can make a PR for this!

**Reply by willmcgugan** [reply]
A PR would be amazing. 👍
